src/daemon/gateway.go

```diff
@@ -541,6 +543,69 @@ func (gw *Gateway) GetWalletSeed(id string, password []byte) (string, error) {
 	return gw.v.Wallets.GetWalletSeed(id, password)
 }
 
+// SaveData saves arbitrary data to data file on disk
+func (gw *Gateway) SaveData(data map[string]interface{}, update bool) error {
```

Comment:
Don't put logic in the gateway. Gateway is only a shim to give the api package an interface for mocking

Reply:
Belongs in its own package. Furthermore, none of this code is tested?

Reply:
The `update` flag makes no sense. If `update` is false it is a no-op?

Reply:
I removed daemon.Gateway; the new way to do this is:

- New package for managing the keystore
- New instance of keystore manager created during coin init
- Instance of keystore manager passed to api.NewGateway
- `Keystorer` interface added to api/gateway.go